apps/asap-server/src/handlers/user/index-handler.ts

```diff
@@ -0,0 +1,62 @@
+import { EventBridgeEvent } from 'aws-lambda';
+import { SquidexGraphql } from '@asap-hub/squidex';
+import {
+  AlgoliaSearchClient,
+  algoliaSearchClientFactory,
+} from '@asap-hub/algolia';
+import logger from '../../utils/logger';
+import Users, { UserController } from '../../controllers/users';
+import { UserEventType } from '../webhooks/webhook-user';
+
+export const indexUserHandler =
+  (
+    userController: UserController,
+    algoliaClient: AlgoliaSearchClient,
+  ): ((
+    event: EventBridgeEvent<UserEventType, SquidexWebhookUserPayload>,
+  ) => Promise<void>) =>
+  async (event: UserIndexEventBridgeEvent): Promise<void> => {
+    logger.debug(`Event ${event['detail-type']}`);
+
+    try {
+      const user = await userController.fetchById(event.detail.payload.id);
+
+      logger.debug(`Fetched user ${user.id}`);
+
+      await algoliaClient.save(user);
+
+      logger.debug(`Saved user ${user.id}`);
+    } catch (e) {
+      if (e?.output?.statusCode === 404) {
+        await algoliaClient.remove(event.detail.payload.id);
```

Comment:
When we receive a delete event, can we be 100% sure that `userController.fetchById` returns 404?

Reply:
so actually the way this handler works is that no matter what event type it receives it will look up the record in Squidex and if 404 is received it assumes that the record was deleted. 

So regardless of the event type we assume that a 404 means record was deleted - are we sure though that this is the only case? yeah i suppose for the 404 we are sure we can remove it from algolia.

Reply:
my concern would be that we receive the delete event but the endpoint still returns a value for the user (e.g. due to cache)
in that case the index would be inconsistent

Reply:
yeah it could happen but on the flip side if we receive a delete event and remove the record without checking that may lead to an even worse outcome